pilot/proxy/envoy/discovery.go

```diff
@@ -659,13 +706,22 @@ func (ds *DiscoveryService) ListListeners(request *restful.Request, response *re
 			errorResponse(methodName, response, http.StatusInternalServerError, "LDS "+err.Error())
 			return
 		}
+
+		// TODO: This does not take into account listeners added by webhook
 		resourceCount = uint32(len(listeners))
-		if resourceCount > 0 {
-			ds.ldsCache.updateCachedDiscoveryResponse(key, resourceCount, out)
+
+		transformedOutput, err := ds.invokeWebhook(fmt.Sprintf("/v1/listeners/unused/%s", svcNode), out)
```

Comment:
I suggest create a single endpoint, and attach a descriptor to each call (listener/route/endpoint, service node, proxy type)

Reply:
any reason why? Using same envoy api eliminates api design issues.. with v2 api, all this falls in place automatically (we post the node identity and the payload as protos or json).

Reply:
xDS HTTP interface is not meant for transforming input to output. This is like overriding an interface method by doing something entirely different.

Reply:
xDS uses HTTP GET, but this always uses POST. It's reusing the resource identifiers, not methods. I like it as a design choice because its simple and unambiguous.

Reply:
Ok, I think it's fair. Exposing the lowest-level xDS hook is the most flexible option.